tests/engine/test_calc_job.py

```diff
@@ -356,36 +356,45 @@ def test_parse_retrieved_folder(self):
 
 
 @pytest.fixture
-def process(aiida_local_code_factory):
+def get_process(aiida_local_code_factory):
     """Instantiate a process with default inputs and return the `Process` instance."""
     from aiida.engine.utils import instantiate_process
     from aiida.manage.manager import get_manager
 
-    inputs = {
-        'code': aiida_local_code_factory('arithmetic.add', '/bin/bash'),
-        'x': orm.Int(1),
-        'y': orm.Int(2),
-        'metadata': {
-            'options': {}
+    def _get_process(inputs=None):
+
+        base_inputs = {
+            'code': aiida_local_code_factory('arithmetic.add', '/bin/bash'),
+            'x': orm.Int(1),
+            'y': orm.Int(2),
+            'metadata': {
+                'options': {}
+            }
         }
-    }
 
-    manager = get_manager()
-    runner = manager.get_runner()
+        if inputs is not None:
+            base_inputs = {**base_inputs, **inputs}
```

Comment:
sure, but then we would have to add two more lines with the `else` clause assigning `extras_inputs = base_inputs`.

Reply:
What? I don't follow. Here you are just checking if there is something in `inputs`/`extra_inputs` and adding that into `base_inputs` and then you continue using `base inputs`, so why would you need to else-assign something to `extra_inputs`?

Maybe your point is that if `inputs` is not descriptive enough, then using `base_inputs` to hold all of the inputs is also a misnomer? And you were thinking of something like this:

```
         if extra_inputs is not None:
            all_inputs = {**base_inputs, **extra_inputs} 
        else:
            all_inputs = base_inputs 
```

So maybe a solution will be then to change the `inputs` provided to the user as `extra_inputs` and the `base_inputs` established by the code to `inputs`? (I would actually use `internal_inputs`, as in internal to this procedure, but I know that this might not sound natural in general)

```
     def _generate_process(extra_inputs=None):

        inputs = {
            'code': aiida_local_code_factory('arithmetic.add', '/bin/bash'),
            ...
            }

         if extra_inputs is not None:
            inputs = {**inputs, **extra_inputs} 
```

Again, super nitpicky; I will approve anyways and you can decide if you want to change this or not.

Reply:
My points was the following. I currently do:
```
def _generate_process(inputs=None):

    base_inputs = {}

    if inputs is not None:
        base_inputs = {**base_inputs, **inputs}

    process = instantiate_process(runner, process_class, **base_inputs)
```
If I were to change `base_inputs = {**base_inputs, **inputs}` to `extra_inputs = {**base_inputs, **inputs}` we have to add two lines.
```
def _generate_process(inputs=None):

    base_inputs = {}

    if inputs is not None:
        extra_inputs = {**base_inputs, **inputs}
    else:
        extra_inputs = base_inputs

    process = instantiate_process(runner, process_class, **extra_inputs)
```
I just prefer the former. Especially given that the variable name `base_inputs` is only used in this very small internal context of `_generate_process` and so I don't think it risks being unclear.

Reply:
Haha, I mean, sure, but what I was suggesting was to change all `inputs` to `extra_inputs`, not adding a new `extra_inputs` variable for the assignment (^ ^ ) : 

```python
def _generate_process(extra_inputs=None):

    base_inputs = {}

    if extra_inputs is not None:
        base_inputs = {**base_inputs, **extra_inputs}

    process = instantiate_process(runner, process_class, **base_inputs)
```

Anyway, thanks for the PR! 👍

Reply:
Ah ok, apologies, I know realize I misinterpreted your very first comment. Thought you didn't like `base_input` being overwritten and used down the line. Thanks for the review